webapp/client/client.jsx

```diff
@@ -1767,6 +1769,30 @@ export default class Client {
             end(this.handleResponse.bind(this, 'getFileInfosForPost', success, error));
     }
 
+    getOpenGraphMetaData(url, success, error) {
+        // There can be only one on going request for one url.
+        // This helps where there are multiple posts with same URL are rendered simultaneously.
+        var self = this;
+        if (!this.openGraphMetaDataOnGoingRequests[url]) {
```

Comment:
In the event there is more than one simultaneous post with the same URL what happens? I don't see success being called for that case, is the post rendering metadata?